code/addons/actions/src/runtime/action.ts

```diff
@@ -54,22 +59,37 @@ export function action(name: string, options: ActionOptions = {}): HandlerFuncti
   };
 
   const handler = function actionHandler(...args: any[]) {
-    // TODO: Enable once codemods are finished
-    // if (options.implicit) {
-    //   const preview =
-    //     '__STORYBOOK_PREVIEW__' in global
-    //       ? (global.__STORYBOOK_PREVIEW__ as PreviewWeb<Renderer>)
-    //       : undefined;
-    //   if (
-    //     preview?.storyRenders.some(
-    //       (render) => render.phase === 'playing' || render.phase === 'rendering'
-    //     )
-    //   ) {
-    //     console.warn(
-    //       'Can not use implicit actions during rendering or playing of a story.'
-    //     );
-    //   }
-    // }
+    if (options.implicit) {
+      const preview =
+        '__STORYBOOK_PREVIEW__' in global
+          ? // eslint-disable-next-line no-underscore-dangle
+            (global.__STORYBOOK_PREVIEW__ as PreviewWeb<Renderer>)
+          : undefined;
+      const storyRenderer = preview?.storyRenders.find(
+        (render) => render.phase === 'playing' || render.phase === 'rendering'
+      );
+
+      if (storyRenderer) {
+        if (window?.FEATURES?.disallowImplicitActionsInRenderV8) {
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          throw new ImplicitActionsDuringRendering({ phase: storyRenderer.phase!, name });
+        } else {
+          console.warn(dedent`
+            We detected that you use an implicit action arg during ${storyRenderer.phase} of your story. 
```

Comment:
Do we also need the story name in this message? Or will it only ever throw on the "current" story? Maybe not with portable stories?

Reply:
So with portable stories, this will never throw. With docs, we don't know which of the rendered stories used the implicit action. Shall I pass down the story id from the arg enhancer maybe?